packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -541,21 +541,38 @@ Future<XcodeBuildResult> buildXcodeProject({
 }
 
 void updateShorebirdYaml(BuildInfo buildInfo, BuildableIOSApp app) {
+  final String resolvedAppName = app.name ?? 'Runner.app';
   final File shorebirdYaml = globals.fs.file(
     globals.fs.path.join(
       app.archiveBundleOutputPath,
       'Products',
       'Applications',
-      app.name ?? 'Runner.app',
+      resolvedAppName,
       'Frameworks',
       'App.framework',
       'flutter_assets',
       'shorebird.yaml',
     ),
   );
   if (!shorebirdYaml.existsSync()) {
+    // Find the closest existing parent of the file.
+    Directory parent = shorebirdYaml.parent;
+
+    int i = 0;
+    // The max depth is just a hard limit to prevent the cli from going to far behind.
+    // This limit should never be reached though, since at least the `Applications` or
+    // `Products` folder should exist, no matter what changed in the app.
+    const int maxDepth = 7;
```

Comment:
Why have the max depth at all?

Reply:
I might be too overcautious here, but I was worried that we end up recurring back too much and end in a user "personal" folder and kind trigger the user that we are invading their files.

This depth makes sure that we don't go past the build folder.

Reply:
Comment should probably say that then. :)